matrix_content_scanner/scanner/scanner.py

```diff
@@ -76,9 +100,42 @@ async def scan_file(
             FileDirtyError if the result of the scan said that the file is dirty, or if
                 the media path is malformed.
         """
+        # Compute the cache key for the media.
```

Comment:
this function seems to conflate the caching and the work somewhat. It might be cleaner to have a separate function which does the download/decrypt/scan etc and returns a `MediaDescription` (or raises a `FileDirtyError`) without worrying about the cache (in other words, the existing `scan_file`) and then build a caching layer on top of that, by inspecting the result and constructing a suitable `CacheEntry`?

(you might need to set a flag on the `FileDirtyError` to say whether it is cacheable or not, but that shouldn't be too bad?)